src/physics/em/detail/MollerBhabhaInteractor.i.hh

```diff
@@ -0,0 +1,219 @@
+        real_type b4   = y122 * y12;
+        real_type b3   = b4 + y122;
+
+        y = x_max * x_max;
+
+        rejection_function_g
+            = 1.0
+              + (y * y * b4 - x_min * x_min * x_min * b3 + y * b2 - x_min * b1)
+                    * beta_sq;
+        do
+        {
+            random[0] = generate_canonical(rng);
+            random[1] = generate_canonical(rng);
+
+            x = x_min * x_max / (x_min * (1.0 - random[0]) + x_max * random[0]);
+            y = x * x;
+            z = 1.0 + (y * y * b4 - x * y * b3 + y * b2 - x * b1) * beta_sq;
+
+        } while (rejection_function_g * random[1] > z);
+    }
+
+    // Calculate secondary kinetic energy
+    real_type secondary_energy = x * inc_energy_.value();
+
+    // Calculate secondary momentum
+    // Same equation as in ParticleTrackView::momentum_sq()
+    real_type secondary_momentum
+        = sqrt(secondary_energy
+               * (secondary_energy + 2.0 * shared_.electron_mass_c_sq));
+
+    // Calculate theta from energy-momentum conservation
+    real_type secondary_cos_theta
+        = secondary_energy * (total_energy + shared_.electron_mass_c_sq)
+          / (secondary_momentum * inc_momentum_.value());
+
+    // Geant says: if (secondary_cos_theta > 1) { secondary_cos_theta = 1; }
+    secondary_cos_theta = std::min(secondary_cos_theta, 1.0);
+    CELER_ASSERT(secondary_cos_theta >= -1.0 && secondary_cos_theta <= 1.0);
+
+    real_type secondary_sin_theta
+        = sqrt((1.0 - secondary_cos_theta) * (1.0 + secondary_cos_theta));
+
+    // Sample phi isotropically
+    UniformRealDistribution<real_type> random_phi(0, 2 * constants::pi);
+    real_type                          secondary_phi = random_phi(rng);
+
+    // Create cartesian direction from the sampled theta and phi
+    Real3 secondary_direction = {secondary_sin_theta * std::cos(secondary_phi),
+                                 secondary_sin_theta * std::sin(secondary_phi),
+                                 secondary_cos_theta};
+
+    // Calculate incident particle final vector momentum
+    Real3 inc_vec_momentum;
+    Real3 secondary_vec_momentum;
+    Real3 inc_exiting_vec_momentum;
+    for (int i : range(3))
+    {
+        inc_vec_momentum[i]       = inc_momentum_.value() * inc_direction_[i];
+        secondary_vec_momentum[i] = secondary_momentum * secondary_direction[i];
+        inc_exiting_vec_momentum[i] = inc_vec_momentum[i]
+                                      - secondary_vec_momentum[i];
+    }
+
+    real_type inc_exiting_momentum
+        = std::sqrt(std::pow(inc_exiting_vec_momentum[0], 2)
+                    + std::pow(inc_exiting_vec_momentum[1], 2)
+                    + std::pow(inc_exiting_vec_momentum[2], 2));
+
+    // Calculate incident particle final direction
+    Real3 inc_exiting_direction;
+    for (int i : range(3))
+    {
+        inc_exiting_direction[i] = inc_exiting_vec_momentum[i]
+                                   / inc_exiting_momentum;
+    }
```

Comment:
You might actually be able to delete this and the above, replace `inc_exiting_vec_momentum` with `inc_exiting_direction`, and call `normalize_direction` on this